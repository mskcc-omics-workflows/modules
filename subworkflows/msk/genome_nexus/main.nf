include { GENOMENEXUS_VCF2MAF      } from '../../../modules/msk/genomenexus/vcf2maf/main'
include { GENOMENEXUS_ANNOTATIONPIPELINE     } from '../../../modules/msk/genomenexus/annotationpipeline/main'

workflow GENOME_NEXUS {

    take:
    ch_vcf // channel: [ val(meta), [ vcf ] ]

    main:

    ch_versions = Channel.empty()


    GENOMENEXUS_VCF2MAF ( ch_vcf )
    ch_versions = ch_versions.mix(GENOMENEXUS_VCF2MAF.out.versions.first())
    GENOMENEXUS_ANNOTATIONPIPELINE( GENOMENEXUS_VCF2MAF.out.maf )


    emit:
    maf      = GENOMENEXUS_VCF2MAF.out.maf           // channel: [ val(meta), [ maf ] ]
    versions = ch_versions                     // channel: [ versions.yml ]
}

