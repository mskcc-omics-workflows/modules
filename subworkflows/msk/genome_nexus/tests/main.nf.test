
// nf-core subworkflows test genome_nexus
nextflow_workflow {

    name "Test Subworkflow GENOME_NEXUS"
    script "../main.nf"
    workflow "GENOME_NEXUS"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/genome_nexus"
    tag "subworkflows_msk"

    tag "genomenexus"
    tag "genomenexus/vcf2maf"
    tag "genomenexus/annotationpipeline"



    test("test - vcf - normal") {

        when {
            workflow {
                """

                input[0] = [ [ id:'test' ], // meta map
                        file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true)
                        ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success}

            )
        }
    }
}
