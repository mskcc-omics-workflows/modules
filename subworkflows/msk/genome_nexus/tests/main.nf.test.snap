{
    "test - vcf - normal": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.maf:md5,548723a34b74a614b654bc8abff855eb"
                    ]
                ],
                "1": [
                    "versions.yml:md5,15a7204e065116da2caa91763fa7ba52"
                ],
                "maf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.maf:md5,548723a34b74a614b654bc8abff855eb"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,15a7204e065116da2caa91763fa7ba52"
                ]
            }
        ],
        "timestamp": "2024-06-18T11:08:08.830895"
    }
}